src/test/java/org/opensearch/geospatial/search/aggregations/bucket/geogrid/GeoHexGridTests.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.search.aggregations.bucket.geogrid;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.lucene.index.IndexWriter;
+import org.hamcrest.MatcherAssert;
+import org.opensearch.common.ParseField;
+import org.opensearch.common.xcontent.ContextParser;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.BaseGeoGrid;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.BaseGeoGridBucket;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGrid;
+import org.opensearch.geospatial.h3.H3;
+import org.opensearch.geospatial.plugin.GeospatialPlugin;
+import org.opensearch.plugins.SearchPlugin;
+import org.opensearch.search.aggregations.Aggregation;
+import org.opensearch.search.aggregations.InternalAggregations;
+import org.opensearch.search.aggregations.ParsedMultiBucketAggregation;
+import org.opensearch.test.InternalMultiBucketAggregationTestCase;
+
+public class GeoHexGridTests extends InternalMultiBucketAggregationTestCase<GeoHexGrid> {
+
+    public static final double LATITUDE_MIN = -90.0;
+    public static final double LATITUDE_MAX = 90.0;
+    public static final double LONGITUDE_MIN = -180.0;
+    public static final double LONGITUDE_MAX = 180.0;
+    public static final int MIN_BUCKET_SIZE = 1;
+    public static final int MAX_BUCKET_SIZE = 3;
+
+    GeoHexGrid createInternalGeoGrid(String name, int size, List<BaseGeoGridBucket> buckets, Map<String, Object> metadata) {
+        return new GeoHexGrid(name, size, buckets, metadata);
+    }
+
+    GeoHexGridBucket createInternalGeoGridBucket(Long key, long docCount, InternalAggregations aggregations) {
+        return new GeoHexGridBucket(key, docCount, aggregations);
+    }
+
+    long longEncode(double lng, double lat, int precision) {
+        return H3.geoToH3(lng, lat, precision);
+    }
+
+    int randomPrecision() {
+        return randomIntBetween(GeoHexHelper.MIN_H3_RES, H3.MAX_H3_RES);
+    }
```

Comment:
any reason for making the package private?

Reply:
> I am not seeing IT tests for the new aggregation?

Added Yaml test.

Reply:
> any reason for making the package private?

Want to start with restrictive access and only open class if it is required. wdyt?

Reply:
Got it. I will add IT tests as well.

Reply:
Will add Integration tests in next PR.